launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/ComponentsImpl.java

```diff
@@ -244,30 +244,36 @@ static final class PollingDataSourceBuilderImpl extends PollingDataSourceBuilder
             implements DiagnosticDescription, DataSourceRequiresFeatureFetcher {
         @Override
         public DataSource build(ClientContext clientContext) {
-            clientContext.getDataSourceUpdateSink().setStatus(
-                    clientContext.isInBackground() ? ConnectionInformation.ConnectionMode.BACKGROUND_POLLING :
+            ClientContextImpl clientContextImpl = ClientContextImpl.get(clientContext);
+            clientContextImpl.getDataSourceUpdateSink().setStatus(
+                    clientContextImpl.isInBackground() ? ConnectionInformation.ConnectionMode.BACKGROUND_POLLING :
                             ConnectionInformation.ConnectionMode.POLLING,
                     null
             );
-            int actualPollIntervalMillis = clientContext.isInBackground() ? backgroundPollIntervalMillis :
+
+            int pollInterval = clientContextImpl.isInBackground() ? backgroundPollIntervalMillis :
                     pollIntervalMillis;
-            int initialDelayMillis;
-            if (clientContext.isInBackground() && Boolean.FALSE.equals(clientContext.getPreviouslyInBackground())) {
```

Comment:
For reviewers: This background/foregrounding distinction is no more!  The only impact foreground/background has is on the polling interval.  Now they are treated the same after that which results in some improvements in behavioral consistency.